modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/resource/v1_0/ObjectFolderResourceImpl.java

```diff
@@ -213,6 +284,125 @@ protected void preparePatch(
+							putObjectDefinitionByExternalReferenceCode(
+								folderItemObjectDefinition.
+									getExternalReferenceCode(),
+								folderItemObjectDefinition);
+
+					com.liferay.object.model.ObjectFolderItem
+						serviceBuilderObjectFolderItem =
+							_objectFolderItemLocalService.
+								createObjectFolderItem(0L);
+
+					serviceBuilderObjectFolderItem.setObjectDefinitionId(
+						objectDefinition.getId());
+					serviceBuilderObjectFolderItem.setObjectFolderId(
+						objectFolderId);
+					serviceBuilderObjectFolderItem.setPositionX(
+						objectFolderItem.getPositionX());
+					serviceBuilderObjectFolderItem.setPositionY(
+						objectFolderItem.getPositionY());
+
+					return serviceBuilderObjectFolderItem;
+				}
+
+				ObjectDefinition objectDefinition =
+					objectDefinitionResource.
+						getObjectDefinitionByExternalReferenceCode(
+							objectFolderItem.
+								getObjectDefinitionExternalReferenceCode());
+
+				if (objectDefinition == null) {
+					return null;
+				}
+
+				_objectDefinitionLocalService.updateObjectFolderId(
+					objectDefinition.getId(), objectFolderId);
+
+				return _objectFolderItemLocalService.updateObjectFolderItem(
+					objectDefinition.getId(), objectFolderId,
+					objectFolderItem.getPositionX(),
+					objectFolderItem.getPositionY());
+			});
+	}
+
+	private List<com.liferay.object.model.ObjectFolderItem>
+		_prepareLinkedObjectItems(
+			List<ObjectFolderItem> folderItems, long objectFolderId,
+			ObjectDefinitionResource objectDefinitionResource) {
+
+		if (folderItems.isEmpty()) {
+			return new ArrayList<>();
+		}
+
+		return transform(
+			folderItems,
+			objectFolderItem -> {
+				ObjectDefinition objectDefinition =
+					objectDefinitionResource.
+						getObjectDefinitionByExternalReferenceCode(
+							objectFolderItem.
+								getObjectDefinitionExternalReferenceCode());
+
+				if (objectDefinition == null) {
+					return null;
+				}
+
+				com.liferay.object.model.ObjectFolderItem
+					serviceBuilderObjectFolderItem =
+						_objectFolderItemLocalService.fetchObjectFolderItem(
+							objectDefinition.getId(), objectFolderId);
+
+				if (serviceBuilderObjectFolderItem == null) {
+					serviceBuilderObjectFolderItem =
+						_objectFolderItemLocalService.createObjectFolderItem(
+							0L);
+
+					serviceBuilderObjectFolderItem.setObjectDefinitionId(
+						objectDefinition.getId());
+					serviceBuilderObjectFolderItem.setObjectFolderId(
+						objectFolderId);
+					serviceBuilderObjectFolderItem.setPositionX(
+						objectFolderItem.getPositionX());
+					serviceBuilderObjectFolderItem.setPositionY(
+						objectFolderItem.getPositionY());
+
+					return serviceBuilderObjectFolderItem;
+				}
+
+				serviceBuilderObjectFolderItem.setPositionX(
+					objectFolderItem.getPositionX());
+				serviceBuilderObjectFolderItem.setPositionY(
+					objectFolderItem.getPositionY());
+
+				return _objectFolderItemLocalService.updateObjectFolderItem(
```

Comment:
Could you try to use the method below instead of use the set and then call the `updateObjectFolderItem`?

```
_objectFolderItemLocalService.updateObjectFolderItem(
					objectDefinition.getId(), objectFolderId,
					objectFolderItem.getPositionX(),
					objectFolderItem.getPositionY());
```